lib/src/typedefs.dart

```diff
@@ -1,7 +1,19 @@
 library react.typedefs;
 
+import 'dart:collection';
+
+import 'package:react/react.dart';
+
 /// Typedef for `react.Component.ref`, which should return one of the following specified by the provided [ref]:
 ///
 /// * `react.Component` if it is a Dart component.
 /// * `Element` _(DOM node)_ if it is a React DOM component.
 typedef dynamic Ref(String ref);
+
+/// Typedef of a transactional [Component.setState] callback.
+///
+/// See: <https://facebook.github.io/react/docs/react-component.html#setstate>
+typedef Map TransactionalSetStateCallback(
+    Map prevState,
+    UnmodifiableMapView props,
```

Comment:
I don't think this should be updated from `Map`, since it limits us in the future from passing in a Map that isn't an `UnmodifiableMapView` (e.g., one that is non-modifiable by some other mechanism that doesn't implement `UnmodifiableMapView`).